### PR TITLE
feat: add `weserv` provider

### PR DESCRIPTION
**Pull request by nandi95**
Most options are in the keymap, the documentation is just a link to the website. That could be improved but frankly that's a lot of work. Maybe someone can put ai to use and summaries the doc for the provider documentation

Recreated as per: https://github.com/nuxt/image/pull/962#issuecomment-1740714317

### CODE REVIEW

File: docs/content/3.providers/weserv.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+title: Weserv
+description: Optimize images with Weserv's dynamic image transformation service.
+---
+
+Weserv offers a wide range of image transformation for all JPEG, PNG, and GIF files.
+
+
+## Modifiers
+
+In addition to `height` and `width`, the Weserv provider numerous modifiers documentation [here](https://images.weserv.nl/docs/quick-reference.html/).
```
Comment (danielroe) [inline docs/content/3.providers/weserv.md:11]:
```suggestion
In addition to `height` and `width`, the Weserv provider numerous modifiers documentation [here](https://images.weserv.nl/docs/quick-reference.html).
```

**Review by danielroe — CHANGES_REQUESTED** [review]
Would you be able to add a few test cases for `weserv` in `playground/providers.ts`, `test/unit/providers.test.ts` and `test/providers.ts`? 🙏

**Comment by nandi95** [comment]
> Would you be able to add a few test cases for `weserv` in `playground/providers.ts`, `test/unit/providers.test.ts` and `test/providers.ts`? 🙏

Screenshot 2023-10-02 at 23 02 05


I can't seem to run the tests. I installed with pnpm and run the test with it but I get the above.

**Comment by danielroe** [comment]
You can try running `pnpm dev:prepare` first.

**Comment by nandi95** [comment]
I know the correct url is generated, and when testing it by pasting it into the browser, I get the result I expected yet. The ssr and no-ssr tests are not working. Can you perhaps help me on this front?

**Comment by kleisauke** [comment]
As per the request made in https://github.com/nuxt/image/pull/962#issuecomment-1706726140, I'd like to provide my feedback:
- Please consider replacing `weserve` with `weserv`.
- Currently, `https://wsrv.nl` is hard-coded, and there's no flexibility to modify it. It's worth noting that many websites host their own solutions using our open-source code.

**Review by danielroe — APPROVED** [review]

**Pull request merged**